lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -457,6 +498,137 @@ def _check_job_arguments(self, jobs, has_parent_job=False):
+
+        # Get run job params from db, and inject parent job id, then run it
+        run_job_params = self._get_run_job_params_from_existing_job(
+            job, user_id=self.sdkmr.user_id
+        )
+
+        # Submit job to job scheduler or fail and not count it as a retry attempt
+        run_job_params[_PARENT_RETRY_JOB_ID] = job_id
+        retry_job_id = self.run(params=run_job_params, as_admin=as_admin)
+
+        # Save that the job has been retried, and increment the count. Notify the parent(s)
+        # 1) Notify the retry_parent that it has been retried
+        # TODO Use and create a method in sdkmr?
+        job.modify(inc__retry_count=1, set__retried=True)
+        # 2) Notify the parent container that it has a new child.. #TODO, maybe move this into multiple retries section
+        if run_job_params[_PARENT_JOB_ID]:
+            parent_job = self.sdkmr.get_job_with_permission(
+                run_job_params[_PARENT_JOB_ID], JobPermissions.WRITE, as_admin=as_admin
+            )
+            parent_job.modify(push__child_jobs=retry_job_id)
+
+            # Should we compare the original and child job to make sure certain fields match,
+        # to make sure the retried job is correctly submitted? Or save that for a unit test?
+        return {"job_id": job_id, "retry_id": retry_job_id}
+
+    @staticmethod
+    def _get_job_input_params_from_existing_job(job_input: JobInput) -> Dict:
+        """
+        Get input level fields.
+        """
+        # Expected fields are found in `_init_job_rec`
+        inputs = dict()
+
+        # Used to iterate over "Job" attributes but in dictionary like lookup
+        inputs_list = [
+            _METHOD,
+            _APP_PARAMS,
+            _SERVICE_VER,
+            _APP_ID,
+            _PARENT_JOB_ID,
+            _WORKSPACE_ID,
+            "requirements",
+        ]
+
+        # Insert field if available
+        for field in inputs_list:
+            if job_input[field]:
+                inputs[field] = job_input[field]
+
+        # _SOURCE_WS_OBJECTS list
+        if job_input[_SOURCE_WS_OBJECTS]:
+            inputs[_SOURCE_WS_OBJECTS] = list(job_input[_SOURCE_WS_OBJECTS])
+
+        # Special Cases: It is OK if meta/narr_cell_info is in params 2x
+        if job_input["narrative_cell_info"]:
+            # For easier testing
+            nci = job_input["narrative_cell_info"]  # type: Meta
+            inputs["meta"] = inputs["narrative_cell_info"] = nci.to_mongo().to_dict()
+
+        # Require a non blank params for job browser or narrative compatibility
+        if _APP_PARAMS not in inputs:
+            inputs[_APP_PARAMS] = {}
+
+        return inputs
+
+    @staticmethod
+    def _get_run_job_params_from_existing_job(job: Job, user_id: str) -> Dict:
+        """
+        Get top level fields from job model to be sent into `run_job`
+        """
+        ji = job.job_input  # type: JobInput
+        job_input = EE2RunJob._get_job_input_params_from_existing_job(ji)
```

Comment:
I'm not sure I understand the advantage of splitting this method out. Doesn't it just mean that instead of doing something like
```
"meta" = job.job_input.narrative_cell_info.to_mongo().to_dict()
```
you're doing
```
"meta": job_input.get(_NARRATIVE_CELL_INFO)
```
?

Seems like a lot of extra code for not much benefit, unless I'm missing something.

Reply:
I created this method to help out when I didn't know all of the items that needed to be extracted, which were optional, required, etc, edge cases. And I didn't know which items had to be copied in both the run_job top level params from the job inputs. However, now I do, so deleting it

Reply:
:+1: muuuuch easier to read